### PR TITLE
Add support for User Account Confirmation actionpacket

### PR DESCRIPTION
**Pull request by sergiohs84**

### CODE REVIEW

**Comment by sergiohs84** [comment]
trigger compilation

File: src/megaapi_impl.cpp [hunk 1]
```diff
@@ -10313,65 +10376,59 @@ void MegaApiImpl::ephemeral_result(error e)
...
         {
-            it->second->setState(MegaTransfer::STATE_FAILED);
-            fireOnTransferFinish(it->second, MegaError(MegaError::API_EACCESS));
+            std::map<int, MegaTransferPrivate *>::iterator it=transferMap.begin();
+            if (it->second)
+            {
+                it->second->setState(MegaTransfer::STATE_FAILED);
+                fireOnTransferFinish(it->second, MegaError(MegaError::API_EACCESS));
+            }
         }
```
Comment (polmr) [inline src/megaapi_impl.cpp]: don't we need
```
            resetTotalDownloads();
            resetTotalUploads();
```
here?

File: bindings/ios/MEGASdk.h [hunk 2]
```diff
@@ -659,7 +659,14 @@ typedef NS_ENUM(NSInteger, HTTPServer) {
  * - [MEGARequest name] - Returns the firstname of the user
  * - [MEGARequest text] - Returns the lastname of the user
  *
- * If this request succeed, a confirmation email will be sent to the users.
+ * Valid data in the MEGARequest object received in onRequestFinish when the error code
+ * is MEGAErrorTypeApiOk:
+ * - [MEGARequest sessionKey] - Returns the session id to resume the process
+ *
+ * If this request succeed, a new ephemeral session will be created for the new user
```
Comment (sergiohs84) [inline bindings/ios/MEGASdk.h]: I've updated the references in `megaapi.h`

File: include/megaapi.h [hunk 3]
```diff
@@ -3930,7 +3930,7 @@ class MegaGlobalListener
         virtual void onNodesUpdate(MegaApi* api, MegaNodeList *nodes);
 
         /**
-         * @brief This function is called when the account has been updated (upgraded/downgraded)
+         * @brief This function is called when the account has been updated (confirmed/upgraded/downgraded)
```
Comment (sergiohs84) [inline include/megaapi.h]: I've added the _confirmed/_ to the other callback, `MegaListener::onAccountUpdate()`

File: include/megaapi.h [hunk 4]
```diff
@@ -5073,18 +5105,41 @@ class MegaApi
...
          * If an account with the same email already exists, you will get the error code
          * MegaError::API_EEXIST in onRequestFinish
          *
          * @param email Email for the account
          * @param base64pwkey Private key calculated with MegaApi::getBase64PwKey
          * @param name Name of the user
          * @param listener MegaRequestListener to track this request
+         *
+         * @deprecated This function is deprecated and will eventually be removed. Instead,
+         * use the new version with firstname and lastname.
```
Comment (sergiohs84) [inline include/megaapi.h]: Added the reference to the specific function.

File: src/megaapi_impl.cpp [hunk 5]
```diff
@@ -9181,26 +9201,69 @@ void MegaApiImpl::fetchnodes_result(error e)
...
             }
         }
+
+        fireOnRequestFinish(request, megaError);
     }
+    else    // TYPE_CREATE_ACCOUNT
+    {
+        if (e != API_OK || request->getParamType() == 1)   // resuming ephemeral session
+        {
+            fireOnRequestFinish(request, e);
```
Comment (sergiohs84) [inline src/megaapi_impl.cpp]: Fixed

**Comment by polmr** [comment]
trigger compilation

**Pull request merged**